src/main/java/com/fasterxml/jackson/annotation/JsonIncludeProperties.java

```diff
@@ -0,0 +1,167 @@
+ * properties (during serialization), or only include processing of
+ * JSON properties read (during deserialization).
+ * <p>
+ * Example:
+ * <pre>
+ * // to only include specified fields from being serialized or deserialized
+ * // (i.e. only include in JSON output; or being set even if they were included)
+ * &#064;JsonIncludeProperties({ "internalId", "secretKey" })
+ * </pre>
+ * <p>
+ * Annotation can be applied both to classes and
+ * to properties. If used for both, actual set will be union of all
+ * includes: that is, you can only add properties to include, not remove
+ * or override. So you can not remove properties to include using
+ * per-property annotation.
+ *
+ * @since 2.12
+ */
+@Target({ElementType.ANNOTATION_TYPE, ElementType.TYPE,
+        ElementType.METHOD, ElementType.CONSTRUCTOR, ElementType.FIELD})
+@Retention(RetentionPolicy.RUNTIME)
+@JacksonAnnotation
+public @interface JsonIncludeProperties
+{
+    /**
+     * Names of properties to include.
+     */
+    public String[] value() default {};
+
+    /*
+    /**********************************************************
+    /* Value class used to enclose information, allow for
+    /* merging of layered configuration settings.
+    /**********************************************************
+     */
+
+    /**
+     * Helper class used to contain information from a single {@link JsonIncludeProperties}
+     * annotation, as well as to provide possible overrides from non-annotation sources.
+     *
+     * @since 2.12
+     */
+    public static class Value implements JacksonAnnotationValue<JsonIncludeProperties>, java.io.Serializable
+    {
+        private static final long serialVersionUID = 1L;
+
+        /**
+         * Default instance has no explicitly included fields
+         */
+        protected final static JsonIncludeProperties.Value ALL = new JsonIncludeProperties.Value(null);
+
+        /**
+         * Name of the properties to include.
+         * Null means that all properties are included, empty means none.
+         */
+        protected final Set<String> _included;
+
+        protected Value(Set<String> included)
+        {
+            _included = included;
+        }
+
+        public static JsonIncludeProperties.Value from(JsonIncludeProperties src)
+        {
+            if (src == null) {
+                return ALL;
+            }
+
+            return new Value(_asSet(src.value()));
+        }
+
+        public static JsonIncludeProperties.Value all()
+        {
+            return ALL;
+        }
+
+        @Override
+        public Class<JsonIncludeProperties> valueFor()
+        {
+            return JsonIncludeProperties.class;
+        }
+
+        public Set<String> getIncluded()
+        {
+            return _included;
+        }
+
+        /**
+         * Mutant factory method to override the current value with an another, merging the included fields.
+         */
```

Comment:
Should explain logic (include if in both (intersection), include if in either (conjunction?)).